src/types.rs

```diff
@@ -8,298 +8,312 @@ use std::fmt;
-    LOG3,
-    LOG4,
-    CREATE,
-    CALL,
-    CALLCODE,
-    RETURN,
-    DELEGATECALL,
-    CREATE2,
-    STATICCALL,
-    REVERT,
-    INVALID,
-    SELFDESTRUCT,
+    STOP = 0,
+    ADD = 1,
+    MUL = 2,
+    SUB = 3,
+    DIV = 4,
+    SDIV = 5,
+    MOD = 6,
+    SMOD = 7,
+    ADDMOD = 8,
+    MULMOD = 9,
+    EXP = 10,
+    SIGNEXTEND = 11,
+    LT = 16,
+    GT = 17,
+    SLT = 18,
+    SGT = 19,
+    EQ = 20,
+    ISZERO = 21,
+    AND = 22,
+    OR = 23,
+    XOR = 24,
+    NOT = 25,
+    BYTE = 26,
+    SHL = 27,
+    SHR = 28,
+    SAR = 29,
+    SHA3 = 32,
+    ADDRESS = 48,
+    BALANCE = 49,
+    ORIGIN = 50,
+    CALLER = 51,
+    CALLVALUE = 52,
+    CALLDATALOAD = 53,
+    CALLDATASIZE = 54,
+    CALLDATACOPY = 55,
+    CODESIZE = 56,
+    CODECOPY = 57,
+    GASPRICE = 58,
+    EXTCODESIZE = 59,
+    EXTCODECOPY = 60,
+    RETURNDATASIZE = 61,
+    RETURNDATACOPY = 62,
+    EXTCODEHASH = 63,
+    BLOCKHASH = 64,
+    COINBASE = 65,
+    TIMESTAMP = 66,
+    NUMBER = 67,
+    DIFFICULTY = 68,
+    GASLIMIT = 69,
+    CHAINID = 70,
+    SELFBALANCE = 71,
+    BASEFEE = 72,
+    POP = 80,
+    MLOAD = 81,
+    MSTORE = 82,
+    MSTORE8 = 83,
+    SLOAD = 84,
+    SSTORE = 85,
+    JUMP = 86,
+    JUMPI = 87,
+    PC = 88,
+    MSIZE = 89,
+    GAS = 90,
+    JUMPDEST = 91,
+    PUSH0 = 95,
+    PUSH1 = 96,
+    PUSH2 = 97,
+    PUSH3 = 98,
+    PUSH4 = 99,
+    PUSH5 = 100,
+    PUSH6 = 101,
+    PUSH7 = 102,
+    PUSH8 = 103,
+    PUSH9 = 104,
+    PUSH10 = 105,
+    PUSH11 = 106,
+    PUSH12 = 107,
+    PUSH13 = 108,
+    PUSH14 = 109,
+    PUSH15 = 110,
+    PUSH16 = 111,
+    PUSH17 = 112,
+    PUSH18 = 113,
+    PUSH19 = 114,
+    PUSH20 = 115,
+    PUSH21 = 116,
+    PUSH22 = 117,
+    PUSH23 = 118,
+    PUSH24 = 119,
+    PUSH25 = 120,
+    PUSH26 = 121,
+    PUSH27 = 122,
+    PUSH28 = 123,
+    PUSH29 = 124,
+    PUSH30 = 125,
+    PUSH31 = 126,
+    PUSH32 = 127,
+    DUP1 = 128,
+    DUP2 = 129,
+    DUP3 = 130,
+    DUP4 = 131,
+    DUP5 = 132,
+    DUP6 = 133,
+    DUP7 = 134,
+    DUP8 = 135,
+    DUP9 = 136,
+    DUP10 = 137,
+    DUP11 = 138,
+    DUP12 = 139,
+    DUP13 = 140,
+    DUP14 = 141,
+    DUP15 = 142,
+    DUP16 = 143,
+    SWAP1 = 144,
+    SWAP2 = 145,
+    SWAP3 = 146,
+    SWAP4 = 147,
+    SWAP5 = 148,
+    SWAP6 = 149,
+    SWAP7 = 150,
+    SWAP8 = 151,
+    SWAP9 = 152,
+    SWAP10 = 153,
+    SWAP11 = 154,
+    SWAP12 = 155,
+    SWAP13 = 156,
+    SWAP14 = 157,
+    SWAP15 = 158,
+    SWAP16 = 159,
+    LOG0 = 160,
+    LOG1 = 161,
+    LOG2 = 162,
+    LOG3 = 163,
+    LOG4 = 164,
+    CREATE = 240,
+    CALL = 241,
+    CALLCODE = 242,
+    RETURN = 243,
+    DELEGATECALL = 244,
+    CREATE2 = 245,
+    STATICCALL = 250,
+    REVERT = 253,
+    INVALID = 254,
+    SELFDESTRUCT = 255,
+}
+
+impl TryFrom<u8> for Opcode {
+    type Error = ();
+
+    fn try_from(x: u8) -> Result<Self, Self::Error> {
+        match x {
+            x if x == Opcode::STOP as u8 => Ok(Opcode::STOP),
```

Comment:
This seems more verbose than the original mapping. I have a hunch there should be a more concise way to do this with the match logic though to avoid all of this repetition. Like using match to check wether the value matches any of the enum values in one line instead of 1 line for each enum variant. what do you think ?